### PR TITLE
2025 08 25 div by one

### PR DESCRIPTION
**Pull request by thedavidmeister**
## Checks

By submitting this for review, I'm confirming I've done the following:
- [ ] made this PR as small as possible
- [ ] unit-tested any new functionality
- [ ] linked any relevant issues or PRs
- [ ] included screenshots (if this involves a front-end change)

## Summary by CodeRabbit

- Bug Fixes
  - Improved division exponent calculation to reduce transient overflow risk in rare edge cases.
  - Adjusted overflow reporting to surface the original exponent in certain overflow paths for clearer diagnostics.

- Tests
  - Added extensive property tests validating division by +1 and -1 across wide exponent ranges, verifying correct value, sign, and normalization.

### CODE REVIEW

File: test/src/lib/implementation/LibDecimalFloatImplementation.div.t.sol [hunk 1]
```diff
@@ -100,6 +100,22 @@ contract LibDecimalFloatImplementationDivTest is Test {
...
+
+        int256 one = 1;
+        for (int256 oneExponent = 0; oneExponent >= -76; --oneExponent) {
+            checkDiv(signedCoefficient, exponent, one, oneExponent, expectedCoefficient, expectedExponent);
+            if (oneExponent == -76) {
+                break;
+            }
+            one *= 10;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (thedavidmeister) [reply test/src/lib/implementation/LibDecimalFloatImplementation.div.t.sol:117]: done

File: test/src/lib/implementation/LibDecimalFloatImplementation.div.t.sol [hunk 2]
```diff
@@ -100,6 +100,39 @@ contract LibDecimalFloatImplementationDivTest is Test {
...
+
+    function testDivByNegativeOneFloat(int256 signedCoefficient, int256 exponent) external pure {
+        exponent = bound(exponent, type(int256).min + 77, type(int256).max - 1);
+        (int256 expectedCoefficient, int256 expectedExponent) =
```
[the remark this answers is not in the document]
Comment (thedavidmeister) [reply test/src/lib/implementation/LibDecimalFloatImplementation.div.t.sol]: this line changed, try again

File: test/src/lib/LibDecimalFloat.div.t.sol [hunk 3]
```diff
@@ -40,4 +40,32 @@
...
+        int256 negativeOne = -1;
+        for (int256 oneExponent = 0; oneExponent >= -65; --oneExponent) {
+            Float result = LibDecimalFloat.div(float, LibDecimalFloat.packLossless(negativeOne, oneExponent));
+            assertTrue(result.eq(float.minus()));
+            if (oneExponent == -65) {
+                break;
+            }
+            negativeOne *= 10;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (thedavidmeister) [reply test/src/lib/LibDecimalFloat.div.t.sol]: is this still relevant?

File: test/src/lib/LibDecimalFloat.div.t.sol [hunk 4]
```diff
@@ -40,4 +40,32 @@ contract LibDecimalFloatDivTest is Test {
...
+        int256 one = 1;
+        for (int256 oneExponent = 0; oneExponent >= -65; --oneExponent) {
+            Float result = LibDecimalFloat.div(float, LibDecimalFloat.packLossless(one, oneExponent));
+            assertTrue(result.eq(float));
+            if (oneExponent == -65) {
+                break;
+            }
+            one *= 10;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (thedavidmeister) [reply test/src/lib/LibDecimalFloat.div.t.sol:56]: this is fuzzed, are you sure these bounds need tightening?
